x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlObjectHandler.java

```diff
@@ -167,6 +169,21 @@ void validateSignature(Signature signature) {
 
         checkIdpSignature(credential -> {
             try {
+                final String signatureAlg = AlgorithmSupport.getKeyAlgorithm(signature.getSignatureAlgorithm());
+                final String keyAlg = credential.getPublicKey().getAlgorithm();
+                if (signatureAlg != null && signatureAlg.equals(keyAlg) == false) {
+                    if (logger.isDebugEnabled()) {
+                        String keyFingerprint = "SHA265:"
+                            + MessageDigests.toHexString(MessageDigests.sha256().digest(credential.getPublicKey().getEncoded()));
+                        logger.debug(
+                            "Skipping [{}] key [{}] because it is not compatible with signature algorithm [{}]",
+                            keyAlg,
+                            keyFingerprint,
+                            signatureAlg
+                        );
+                    }
+                    return false;
+                }
```

Comment:
This bit is from https://github.com/elastic/elasticsearch/pull/98119 . Without it there are test failures such as https://gradle-enterprise.elastic.co/s/6hj4qm6ntsbpc/tests/task/:x-pack:plugin:security:test/details/org.elasticsearch.xpack.security.authc.saml.SamlAuthenticatorTests/testSigningWhenIdpHasMultipleKeys?top-execution=1 .
This shows that the internals of the lib changed in between the version updates, so there is a certain risk in the update.